articles/ToDoList-with-Flask-Firebase/index.md

```diff
@@ -0,0 +1,241 @@
+   def add_task():
+       if request.method == 'POST' and request.form['Task'] != '':
+           print(request.form['Task'])
+           db.push(request.form['Task'])
+       return render_index()
+   ```
+   
+   Here, we take the task that was submitted by index.html and push it into the database (create a new child node), and reload the page.
+   
+8. Let us also finish our delete and update functions.
+
+   ```python
+   @app.route('/delete_task',methods=['POST'])
+   def delete_task():
+       if request.method == 'POST' and request.form['delete'] != '':
+           print(request.form['delete'])
+           db.child(request.form['delete']).remove()
+       return render_index()
+   ```
+
+   Upon clicking delete, we are passing the key of the task to Firebase and deleting it.
+
+   ```python
+   @app.route('/update_task',methods=['POST'])
+   def update_task():
+       if request.method == 'POST' and request.form['update'] != '':
+           print(request.form['update'])
+           db.update({request.form['update']:request.form['task']})
+       return render_index()
+   ```
+
+   Here, we take the change that was made to the task, and we update that task in the database.
+
+   ### Step 4: Updating our HTML with Jinja
+
+   Jinja is a templating language for Python and will help the Python-based servers, like Flask, communicate with the HTML files. We will make use of this in our application to pass data from server.py to index.html.
+
+   To display all the tasks that have been added, we are passing the dictionary ‘todo’ in render_index() to index.html. This will be received and used to generate as many tasks as there are in the dictionary. These tasks are accompanied by the update and delete buttons.
+
+   So the code will be modified to look like this -
+
+   ```html
+   <!-- Task List -->
+   <div style="background-color: lightcyan">
+     {% for l in todo %}
+           <fieldset>
+             <form id="deleteTask" action="{{ url_for('delete_task') }}" method="post">
+             	<button  name="delete" value={{ l }} type="submit"  style="float: right;">Delete</button>
+             </form>
+             <form id="updateTask" action="{{ url_for('update_task') }}" method="post">
+             	<input type="text" name="task" value='{{ todo[l] }}' required>
+             	<br>
+             	<button  name="update" value={{ l }} type="submit"  style="float: left;">Update</button>
+             </form>
+           </fieldset>
+   {% endfor %}
+       </div>
+       <!-- Task List -->
+   ```
+
+   We have a for loop to iterate through all the elements in ‘todo’, and then we set the button values to ‘l’ as it represents the key, ie, the ID used to store the task in the database. This helps us identify which task needs to be updated or deleted. And the input value is set as ‘todo[l]’, which contains the value of the task, ie, the task itself. Jinja templating is done within double curly brackets, and is similar to Python code.
```

Comment:
consider using the methods I highlighted above to make `todo`, `l`, `todo[l]`  more prominent.